python/adbc_driver_manager/adbc_driver_manager/dbapi.py

```diff
@@ -926,6 +927,44 @@ def fetch_df(self) -> "pandas.DataFrame":
             )
         return self._results.fetch_df()
 
+    def fetch_record_batch(self, rows_per_batch: int) -> Optional["_BatchIterator"]:
```

Comment:
This should be called `fetch_record_batch_reader`, and should return a pyarrow.RecordBatchReader, IMO

Reply:
The intent of the issue was to get the PyArrow record batch reader, which avoids going through Python objects like BatchIterator

Reply:
The implementation can just be `return self._results._reader` IMO, or possibly we can wrap the reader to finagle the lifetime (but I'm not sure it's worth doing it right away)

Reply:
Ah, ok, I see...so to mimic DuckDB, this should just return RecordBatchReader, and we should remove the `read_next_batch` method below, since that's a method of the RecordBatchReader